packages/components/src/hooks/use-previous.js

```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 
-export const usePreviousState = value => {
+export const usePrevious = value => {
```

Comment:
Do we really need to rename this hook? I think it will just make it harder for developers to understand what this hook does without a noun in the name.

Reply:
Actually I looked into a few articles before changing. Most of them did use `usePrevious`. I removed state as it's not just for states and IMO it can be more of confusion later. If you really think it's gonna be hard to understand we can change it to something like `usePreviousValue` but I believe it's OK :smile:

Reply:
Hmm, since it's a general hook used not only for a state than I believe `usePrevious` is a good name, so let's stick to it. My concern was mainly about a confusion that can happen in case if we have other hooks with similar names (e.g. `usePreviousLocation`), but since this is a general one, it should not be an issue.

Reply:
@salar-fs , yeah, I don't really mind the name because it's supposedly to replicate the `prevProps` or `prevState` in `componentDidMount` lifecycle, but I prefer a better name for this hook instead of a hanging definition. 

I would prefer something like`usePreviousRefValues` rather than just `usePrevious` but that just's my own nitpick.
We can keep it as is, as you both are unanimous in agreement.

Reply:
Thanks @zaki-binary :pray: 
Yes I updated all other instances as well.